From c39f3d02776a6312dfd14a21dc32c8f262a0bb56 Mon Sep 17 00:00:00 2001
From: Ludovic Barre <ludovic.barre@foss.st.com>
Date: Wed, 27 Nov 2024 11:08:20 +0100
Subject: [PATCH 1437/1695] product/tfm-stm32mp2: update for tfm v2.1.0

Update for tfm v2.1.0
- new path STM_FAMILLY_DIR
- add cmsis and tfm_config link
Fix indentation

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 364db1e22789f7d0250891aba83686a75e7c4e92 product/tfm-stm32mp2: update for tfm v2.1.0]
Change-Id: I97197e6e461321bf513ba426c83f6efbddcdb889
Signed-off-by: Ludovic Barre <ludovic.barre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/SCP-firmware/+/427908
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 .../product/tfm-stm32mp2/fw/CMakeLists.txt    | 25 ++++++++---------
 .../module/stm32_pd/CMakeLists.txt            | 26 +++++++++++------
 .../module/tfm_clock/CMakeLists.txt           | 23 ++++++++++-----
 .../module/tfm_mbx/CMakeLists.txt             | 28 ++++++++++++-------
 4 files changed, 63 insertions(+), 39 deletions(-)

diff --git a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/CMakeLists.txt b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/CMakeLists.txt
index e0b203445..dc3c71dc1 100644
--- a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/CMakeLists.txt
+++ b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/CMakeLists.txt
@@ -13,20 +13,17 @@ add_library(${SCP_FIRMWARE_TARGET})
 
 # cmake-lint: disable=E1122
 
-target_include_directories(
-    ${SCP_FIRMWARE_TARGET}
-    PUBLIC  ${CMAKE_CURRENT_SOURCE_DIR}/../include
-            ${SCP_TFM_PLAT_DIR}/native_driver/include
-            ${SCP_TFM_PLAT_DIR}/../devicetree/include
-            ${SCP_TFM_PLAT_DIR}/native_driver
-            ${CMAKE_BINARY_DIR}/generated/devicetree/platform_s
-            ${STM_SOC_DIR}/secure_fw/partitions/scp
+target_include_directories(${SCP_FIRMWARE_TARGET}
+	PUBLIC
+		${CMAKE_CURRENT_SOURCE_DIR}/../include
+		${SCP_TFM_PLAT_DIR}/native_driver/include
+		${SCP_TFM_PLAT_DIR}/../devicetree/include
+		${SCP_TFM_PLAT_DIR}/native_driver
+		${CMAKE_BINARY_DIR}/generated/devicetree/platform_s
+		${STM_FAMILLY_DIR}/secure_fw/partitions/scp
 )
 
-target_sources(
-    ${SCP_FIRMWARE_TARGET}
-    PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/config_all.c"
+target_sources(${SCP_FIRMWARE_TARGET}
+	PRIVATE
+		${CMAKE_CURRENT_SOURCE_DIR}/config_all.c
 )
-
-
-
diff --git a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/stm32_pd/CMakeLists.txt b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/stm32_pd/CMakeLists.txt
index 5037a936c..5fbcb580d 100644
--- a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/stm32_pd/CMakeLists.txt
+++ b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/stm32_pd/CMakeLists.txt
@@ -8,18 +8,28 @@
 add_library(${SCP_MODULE_TARGET} SCP_MODULE)
 
 target_include_directories(${SCP_MODULE_TARGET}
-                           PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
+	PUBLIC
+		${CMAKE_CURRENT_SOURCE_DIR}/include
+)
 
 # These include directories are relative to SCP-firmware submodule position in
 # tfm-os tree
 target_include_directories(${SCP_MODULE_TARGET}
-                           PUBLIC ${SCP_TFM_PLAT_DIR}/native_driver/include
-                                  ${SCP_TFM_PLAT_DIR}/../devicetree/include
-                                  ${CMAKE_BINARY_DIR}/generated/devicetree/platform_s
-                                  ${SCP_TFM_DIR}/secure_fw/include)
+	PUBLIC
+		${SCP_TFM_PLAT_DIR}/native_driver/include
+		${SCP_TFM_PLAT_DIR}/../devicetree/include
+		${CMAKE_BINARY_DIR}/generated/devicetree/platform_s
+		${SCP_TFM_DIR}/secure_fw/include
+)
 
 
-target_sources(
-    ${SCP_MODULE_TARGET} PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/src/mod_stm32_pd.c")
+target_sources(${SCP_MODULE_TARGET}
+	PRIVATE
+		${CMAKE_CURRENT_SOURCE_DIR}/src/mod_stm32_pd.c
+)
 
-target_link_libraries(${SCP_MODULE_TARGET} PUBLIC module-power-domain)
+target_link_libraries(${SCP_MODULE_TARGET}
+	PUBLIC
+		module-power-domain
+		tfm_config
+)
diff --git a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_clock/CMakeLists.txt b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_clock/CMakeLists.txt
index 319139685..8f7a71581 100644
--- a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_clock/CMakeLists.txt
+++ b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_clock/CMakeLists.txt
@@ -8,18 +8,27 @@
 add_library(${SCP_MODULE_TARGET} SCP_MODULE)
 
 target_include_directories(${SCP_MODULE_TARGET}
-                           PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
+	PUBLIC
+		${CMAKE_CURRENT_SOURCE_DIR}/include
+)
 
 # These include directories are relative to SCP-firmware submodule position in
 # tfm-os tree
 target_include_directories(${SCP_MODULE_TARGET}
-                           PUBLIC ${SCP_TFM_PLAT_DIR}/native_driver/include
-                                  ${SCP_TFM_PLAT_DIR}/../devicetree/include
-                                  ${CMAKE_BINARY_DIR}/generated/devicetree/platform_s
-                                  ${SCP_TFM_DIR}/secure_fw/include)
+	PUBLIC
+		${SCP_TFM_PLAT_DIR}/native_driver/include
+                ${SCP_TFM_PLAT_DIR}/../devicetree/include
+		${CMAKE_BINARY_DIR}/generated/devicetree/platform_s
+		${SCP_TFM_DIR}/secure_fw/include
+)
 
 target_sources(${SCP_MODULE_TARGET}
-               PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/src/mod_tfm_clock.c")
+        PRIVATE
+		${CMAKE_CURRENT_SOURCE_DIR}/src/mod_tfm_clock.c
+)
 
 target_link_libraries(${SCP_MODULE_TARGET}
-                      PRIVATE module-clock)
+	PRIVATE
+		module-clock
+		tfm_config
+)
diff --git a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_mbx/CMakeLists.txt b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_mbx/CMakeLists.txt
index ca2a94165..2e7d23af3 100644
--- a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_mbx/CMakeLists.txt
+++ b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_mbx/CMakeLists.txt
@@ -8,21 +8,29 @@
 add_library(${SCP_MODULE_TARGET} SCP_MODULE)
 
 target_include_directories(${SCP_MODULE_TARGET}
-                           PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
+	PUBLIC
+		${CMAKE_CURRENT_SOURCE_DIR}/include
+)
 
 # These include directories are relative to SCP-firmware submodule position in
 # tfm tree
 target_include_directories(${SCP_MODULE_TARGET}
-                           PUBLIC ${SCP_TFM_PLAT_DIR}/core/include/
-                                  {SCP_TFM_PLAT_DIR}/lib/libutils/ext/include/
-                                  ${SCP_TFM_PLAT_DIR}/lib/libutee/include/
-                                  ${SCP_TFM_DIR}/platform/ext/cmsis)
+	PUBLIC
+		${SCP_TFM_PLAT_DIR}/core/include
+		{SCP_TFM_PLAT_DIR}/lib/libutils/ext/include
+		${SCP_TFM_PLAT_DIR}/lib/libutee/include
+		${SCP_TFM_DIR}/platform/ext/cmsis
+)
 
 target_link_libraries(${SCP_MODULE_TARGET}
-                      PRIVATE
-                              module-msg-smt
-                              module-scmi
-                              module-tfm-smt)
+	PRIVATE
+		module-msg-smt
+		module-scmi
+		module-tfm-smt
+		cmsis
+)
 
 target_sources(${SCP_MODULE_TARGET}
-               PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/src/mod_tfm_mbx.c")
+	PRIVATE
+		${CMAKE_CURRENT_SOURCE_DIR}/src/mod_tfm_mbx.c
+)
-- 
2.47.2

